src/vsg/threading/OperationThreads.cpp

```diff
@@ -30,10 +30,26 @@ OperationThreads::OperationThreads(uint32_t numThreads, ref_ptr<Active> in_activ
             }
         }
     };
+    auto run = [](ref_ptr<OperationQueue> q, ref_ptr<Active> a) {
```

Comment:
Why have two run functions? The new one will essentially be spin lock and just sit their consuming 
CPU resources regardless of whether there is any load or not.  

The implementation of the take_when_available is pretty efficient and shouldn't result in threads waiting long times before waking up to do work.

If the general solution is more efficient on CPU resources and work fast enough then it's something we should strongly prefer over spin locks.  If users want to do something undesirable then I would generally try to steer them in the direction of doing something that is generally better.

If doing the right then for normal usage and keeping the code simple is something I'd use as the default, adding complexity for very niche usage cases is something we should avoid, especially if it can be misused by developers who might not realized what they are doing can be lead to significant inefficiencies.

Reply:
it was mainly for test purpose but I admit I haven't seen any difference